src/main/scala/com/thenewmotion/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -124,7 +124,7 @@ class ConnectionActor(
     log.debug("{} closing broken connection {}", self.path, connection)
     closeIfOpen(connection)
 
-    self ! Connect
+    setTimer(reconnectTimer, Connect, reconnectionDelay / 3, repeat = false)
```

Comment:
I add a small delay before the ConnectionActor try to re-establish the connection. This will cause the `ConnectionActor` stays in `Disconnected` state when it receive the `ProvideChannel` message which sent by children actors acquiring for new channel.
Then when the connection recovered, ConnectionActor will create new channels for each of it's children. 
So, children got new channels, and no duplicated channel created, no conflicts.